File: app/src/main/java/in/smslite/adapter/SMSAdapter.java
package in.smslite.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import in.smslite.R;
import in.smslite.activity.MainActivity;
import in.smslite.db.Message;
import in.smslite.viewHolder.SMSViewHolder;

/**
 * Created by rahul1993 on 11/15/2017.
 */

public class SMSAdapter extends RecyclerView.Adapter<SMSViewHolder> {
    public  List<Message> messages;
    private final static int READ = 0;
    private final static int UNREAD = 1;
    public  List<Message> selectedItemAdapter = new ArrayList<>();
    public  List<Message> listOfItemAdapter = new ArrayList<>();

  public SMSAdapter(List<Message> messages, List<Message> selectedItemAdapter, List<Message> listOfItemAdapter){
    this.messages = messages;
    this.listOfItemAdapter = listOfItemAdapter;
    this.selectedItemAdapter =selectedItemAdapter;
  }

  @Override
  public SMSViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    if (viewType == READ) {
      View viewHolder = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_read_sms, parent, false);
      return new SMSViewHolder(viewHolder, parent.getContext());
    }
    else {
      View viewHolder = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_unread_sms, parent, false);
      return new SMSViewHolder(viewHolder, parent.getContext());
    }
  }

  @Override
  public void onBindViewHolder(SMSViewHolder holder, int position) {
    if(selectedItemAdapter.contains(listOfItemAdapter.get(position))){
      holder.setBackgroundColor();
      holder.setDividerLineInvisible();
    } else {
      holder.setBackgroundColorWhite();
      holder.setDividerLineVisible();
    }
    holder.setAddress(messages.get(position).getAddress());
    holder.setSummary(messages.get(position).getBody());
    holder.setTime(messages.get(position).getTimestamp());
  }

  @Override
  public int getItemCount() {
    return messages.size();
  }

  @Override
  public int getItemViewType(int position){
    return (messages.get(position).read) ? READ : UNREAD;
  }

  public void setMessage(List<Message> messages){
    this.messages = messages;
    notifyDataSetChanged();
  }
}

File: app/src/main/java/in/smslite/adapter/SMSAdapter.kt
package `in`.smslite.adapter

import `in`.smslite.R
import `in`.smslite.db.Message
import `in`.smslite.viewHolder.SMSViewHolder
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import java.util.*

/**
 * Created by rahul1993 on 11/15/2017.
 */
class SMSAdapter(var messages: List<Message>, selectedItemAdapter: List<Message>, listOfItemAdapter: List<Message>) : RecyclerView.Adapter<SMSViewHolder>() {
    var selectedItemAdapter: List<Message> = ArrayList()
    var listOfItemAdapter: List<Message> = ArrayList()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SMSViewHolder {
        return if (viewType == READ) {
            val viewHolder: View = LayoutInflater.from(parent.context).inflate(R.layout.card_read_sms, parent, false)
            SMSViewHolder(viewHolder, parent.context)
        } else {
            val viewHolder: View = LayoutInflater.from(parent.context).inflate(R.layout.card_unread_sms, parent, false)
            SMSViewHolder(viewHolder, parent.context)
        }
    }

    override fun onBindViewHolder(holder: SMSViewHolder, position: Int) {
        if (selectedItemAdapter.contains(listOfItemAdapter[position])) {
            holder.setBackgroundColor()
            holder.setDividerLineInvisible()
        } else {
            holder.setBackgroundColorWhite()
            holder.setDividerLineVisible()
        }
        holder.setAddress(messages[position].getAddress())
        holder.setSummary(messages[position].getBody())
        holder.setTime(messages[position].getTimestamp())
    }

    override fun getItemCount(): Int {
        return messages.size
    }

    override fun getItemViewType(position: Int): Int {
        return if (messages[position].read) READ else UNREAD
    }

    fun setMessage(messages: List<Message>) {
        this.messages = messages
        notifyDataSetChanged()
    }

    companion object {
        private const val READ = 0
        private const val UNREAD = 1
    }

    init {
        this.listOfItemAdapter = listOfItemAdapter
        this.selectedItemAdapter = selectedItemAdapter
    }
}